packages/block-library/src/media-text/edit.js

```diff
@@ -148,9 +153,35 @@ function MediaTextEdit( { attributes, isSelected, setAttributes } ) {
 		rel,
 		verticalAlignment,
 		allowedBlocks,
+		useFeaturedImage,
 	} = attributes;
 	const mediaSizeSlug = attributes.mediaSizeSlug || DEFAULT_MEDIA_SIZE_SLUG;
 
+	const [ featuredImage ] = useEntityProp(
+		'postType',
+		postType,
+		'featured_media',
+		postId
+	);
+
+	const featuredImageMedia = useSelect(
+		( select ) =>
+			featuredImage &&
+			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
+		[ featuredImage ]
+	);
+
+	const toggleUseFeaturedImage = () => {
+		setAttributes( {
+			imageFill: false,
+			mediaType: 'image',
+			mediaId: featuredImage,
```

Comment:
this is different compared to the cover block - we save the actual URL of the featured image here, and we should not, the idea is to flag the block as using the featured imahe but the media id and url should be undefined.

Reply:
When I set `mediaUrl` to undefined, selecting the "Use featured image" option does not work.

Reply:
The only way I could think of, to work around not saving the actual URL, was to introduce an index.php file
that fetches and displays the featured image data on the front.

But this lead to other issues, because there are several block attributes that are not available when registering the block using PHP, in addition to the text-based content, the inner blocks. I am not sure if I should try to work with the `innerHTML` / content, try to use WP_HTML_Tag_Processor, or change the attributes.

Reply:
I believe we _have_ to make the block dynamic in order for it to support featured image. I suggest lean in WP_HTML_Tag_Processor and alter what the save function provides while rendering.

Reply:
I keep thinking that I can *get content* with the tag processor but no  :) 
It doesn't quiet do what I want it to do.